ledger/complete/wal/checkpointer.go

```diff
@@ -769,3 +773,43 @@ func readCheckpointV5(f *os.File) ([]*trie.MTrie, error) {
 
 	return tries, nil
 }
+
+// requestDropFromOSFileCache requests the specified file
+// be dropped from OS file cache.
+// CAUTION: Returns nil without doing anything if GOOS != linux.
+func requestDropFromOSFileCache(fileName string) error {
+	if runtime.GOOS != "linux" {
+		return nil
+	}
+
+	// Try using /bin/dd (Debian, Ubuntu, etc.)
+	cmdFileName := "/bin/dd"
+
+	// If /bin/dd isn't found, then try /usr/bin/dd (OpenSUSE Leap, etc.)
+	_, err := os.Stat(cmdFileName)
+	if os.IsNotExist(err) {
+		cmdFileName = "/usr/bin/dd"
+		_, err := os.Stat(cmdFileName)
+		if os.IsNotExist(err) {
+			return fmt.Errorf("required program dd not found in /bin/ and /usr/bin/")
+		}
+	}
+
+	// Remove some special chars from fileName just in case.
+	// Regex would be shorter but not as easy to read.
+	s := strings.ReplaceAll(fileName, " ", "")
+	s = strings.ReplaceAll(s, ";", "")
+	s = strings.ReplaceAll(s, "$", "")
+	s = strings.ReplaceAll(s, "|", "")
+	s = strings.ReplaceAll(s, ">", "")
+	s = strings.ReplaceAll(s, "<", "")
+	s = strings.ReplaceAll(s, "*", "")
+
+	_, err = os.Stat(s)
+	if os.IsNotExist(err) {
+		return fmt.Errorf("sanitized filename %s does not exist", s)
+	}
+
+	cmd := exec.Command(cmdFileName, "if="+s, "iflag=nocache", "count=0")
```

Comment:
Better to INFO log this operation

Reply:
Great suggestion! 👍   Added logging in 3bf1320f82fe6abaff0aeac682c213970e1fdc7b.

Unfortunately, adding it required more changes than anticipated because the caller didn't have access to a logger.

Reply:
I see. Thanks for adding it. Actually it's not required to pass the logger all the way down. To me, it's also OK to simply [use global logger like this](https://github.com/onflow/flow-go/blob/petera/execution-data-requester/cmd/bootstrap/cmd/check_machine_account.go#L91)